From 72d39f351e56d4a042b51251e7344d750afbbeb0 Mon Sep 17 00:00:00 2001
From: segfault <segfault@riseup.net>
Date: Sun, 23 Jan 2022 20:06:55 +0100
Subject: [PATCH] Support opening new tabs in running Tor Browser instance

This allows using Tor Browser as the default browser without getting a
"Tor Browser is already running but is not responding" error.
---
 torbrowser_launcher/launcher.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/torbrowser_launcher/launcher.py b/torbrowser_launcher/launcher.py
index 55e5263..d039f53 100644
--- a/usr/lib/python3/dist-packages/torbrowser_launcher/launcher.py
+++ b/usr/lib/python3/dist-packages/torbrowser_launcher/launcher.py
@@ -485,9 +485,10 @@ class Launcher(QtWidgets.QMainWindow):
             return
 
         # Run Tor Browser
-        subprocess.call(
-            [self.common.paths["tbb"]["start"]], cwd=self.common.paths["tbb"]["dir_tbb"]
-        )
+        cmd = [self.common.paths["tbb"]["start"], "--allow-remote"]
+        if len(sys.argv) > 1:
+            cmd += ["--new-tab", sys.argv[1]]
+        subprocess.call(cmd, cwd=self.common.paths["tbb"]["dir_tbb"])
         sys.exit(0)
 
     # Start over and download TBB again
-- 
2.34.1

